pkg/controllers/migration/resource/controller.go

```diff
@@ -69,7 +69,7 @@ func (c *Controller[T]) Reconcile(ctx context.Context, req reconcile.Request) (r
 	}))
 	// update annotation on CR
 	if !equality.Semantic.DeepEqual(stored, o) {
-		if err := c.kubeClient.Patch(ctx, o, client.MergeFromWithOptions(stored.(client.Object), client.MergeFromWithOptimisticLock{})); err != nil {
+		if err := c.kubeClient.Patch(ctx, o, client.MergeFrom(stored.(client.Object))); client.IgnoreNotFound(err) != nil {
```

Comment:
We still need to do an optimistic lock here to avoid reserializing parts of the NodeClaim spec, right?

Reply:
Shouldn't, right? It's just an annotation update?

Reply:
Sorry, I got my wires crossed on the issues we were running into. Since it's still a patch we're good.